2016/proposals/blessing-chick.md

```diff
@@ -0,0 +1,64 @@
+# A scikit-bio-based bioinformatics file format converter
+
+## Abstract
+In bioinformatics, there are many defined file formats that represent very similar data. scikit-bio has a powerful I/O framework that enables users to load diverse formats into their relevant in-memory representations, regardless on input file format. It would be useful for many bioinformatics end-users (who are often not comfortable working with APIs) to have a file format converter for formats supported in scikit-bio. The goal of this project is to develop a scikit-bio Python API for file format conversions and then develop a simple web based conversion tool to for easy file upload and conversion. This project would also likely involve the development of additional file format readers and writers for scikit-bio, to increase the power of this application and of scikit-bio.
+
+## Technical Details
+scikit-bio is an open-source python package providing data structures, algorithms, and educational resources for bioinformatics.  It provides  a powerful I/O framework  called the I/O registry which enables users to load diverse formats into their relevant in-memory representations, regardless on input file format. The scikit-bio I/O registry provides a single entry point to all I/O using a simple procedural interface. Additionally, the registry dynamically generates an equivalent object-oriented API for any scikit-bio object that can be serialized or deserialized (i.e., written to or read from a file). Finally, the registry supports automatic file format detection via file sniffers. In this project, we will develop an API for file conversion based on this framework and in the second part of the project, we will develop a file conversion tool using our API. The file conversion tool will also allow us to upload unknown file formats and use in build sniffers from scikit-bio  to auto detect the file format before conversion.
+We will use the Django REST framework in this project to develop the API.
```

Comment:
Why Django REST framework? I think that will be a lot easier to go with Flask. I don't think that you need database, authentication, ...